### DISCUSSION TITLE
Long running jobs - JWT Token expiration

### DISCUSSION THREAD

**Discussion by paulwiesi** [supabase/supabase#14459] [Questions] [answered]
I have multiple long running jobs on an express server and I am using Row level security inside supabase for tenant & user seperation. I am currently passing the jwt token of the user to this jobs (Insert new data, change data,...)

My problem is, these jobs could take longer then the lifetime of the JWT Token (e.g. the jwt token was already near its expire date) and generate an error. I don't want to use the service_role, because it would circumvent RLS and errors could creep in more easily. (e.g. have to add .eq("tenant_id", id) to every request) 
What would be the best solution for this ?

**Comment by GaryAustin1** [comment, marked as answer]
Mint a copy of the jwt with whatever expiration you want on the server.  You can use any jwt library and the Supabase jwt secret to do that.
